tests/unit/test_hnswlib.cpp

```diff
@@ -39,6 +39,65 @@ TEST_F(HNSWLibTest, hnswlib_vector_add_test) {
     ASSERT_EQ(VecSimIndex_IndexSize(index), 1);
     VecSimIndex_Free(index);
 }
+
+TEST_F(HNSWLibTest, hnswlib_blob_sanity_test) {
+    size_t dim = 4;
+    size_t bs = 1;
+#define ASSERT_HNSW_BLOB_EQ(id, blob)                                                              \
+    do {                                                                                           \
+        void *v = reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getDataByInternalId(id);   \
+        ASSERT_FALSE(memcmp(v, blob, sizeof(blob)));                                               \
```

Comment:
Is sizeof(blob) return the actual length or the size of the pointer?

Reply:
Size of the entire blob in bytes